backend/src/data/repositories/user/user.repository.ts

```diff
@@ -18,27 +21,51 @@ class User {
   public async getAll({
     page,
     count,
-  }: EntityPaginationRequestQueryDto): Promise<EntityPagination<UserM>> {
-    const result = await this.#UserModel.query().page(page, count);
+  }: EntityPaginationRequestQueryDto): Promise<
+    EntityPagination<UsersGetResponseDto>
+  > {
+    const result = await this.#UserModel
+      .query()
+      .select('users.id', 'users.createdAt', 'email', 'fullName')
+      .joinRelated('userDetails')
+      .page(page, count)
+      .castTo<UserWithDetails>();
```

Comment:
```suggestion
      .castTo<EntityPagination<UsersGetResponseDto>>();
```
don't forget to delete UserWithDetails